src/lib/strong_typedef.hpp

```diff
@@ -20,7 +20,7 @@
 
 #define STRONG_TYPEDEF(T, D)                                                                                      \
   namespace opossum {                                                                                             \
-  struct D : boost::totally_ordered1<D, boost::totally_ordered2<D, T>> {                                          \
+  struct D : boost::totally_ordered1<D> {                                                                         \
```

Comment:
Can you explain to me why you changed that?

Reply:
For some reason, clang now thinks that there are two possibilities for `operator==` - the one defined by boost and the one if using the user-defined conversion to the base type. Removing this makes sure that comparisons to the base type always use the second path.